pkg/provider/azure_backoff.go

```diff
@@ -272,7 +275,27 @@ func (az *Cloud) ListLB(service *v1.Service) ([]network.LoadBalancer, error) {
 		return nil, rerr.Error()
 	}
 	klog.V(2).Infof("LoadBalancerClient.List(%v) success", rgName)
-	return allLBs, nil
+
+	agentPoolLBs := make([]network.LoadBalancer, 0)
+	agentPoolVMSetNames, err := az.VMSet.GetAgentPoolVMSetNames(nodes)
+	if err != nil {
+		return nil, fmt.Errorf("ListAgentPoolLBs: failed to get agent pool vmSet names: %w", err)
+	}
+	agentPoolVMSetNamesSet := sets.NewString()
+	if agentPoolVMSetNames != nil && len(*agentPoolVMSetNames) > 0 {
+		for _, vmSetName := range *agentPoolVMSetNames {
+			agentPoolVMSetNamesSet.Insert(vmSetName)
+		}
+	}
+
+	for _, lb := range allLBs {
+		vmSetNameFromLBName := az.mapLoadBalancerNameToVMSet(to.String(lb.Name), clusterName)
+		if agentPoolVMSetNamesSet.Has(vmSetNameFromLBName) {
+			agentPoolLBs = append(agentPoolLBs, lb)
+		}
+	}
```

Comment:
Filtered out lbs not managed by provider azure or not associated to an agent pool vmSet.